iroh/src/node/builder.rs

```diff
@@ -125,29 +63,19 @@ async fn spawn_docs<S: iroh_blobs::store::Store>(
 ///
 /// [number 0]: https://n0.computer
 #[derive(derive_more::Debug)]
-pub struct Builder<D>
-where
-    D: Map,
-{
+pub struct Builder {
     storage: StorageConfig,
     addr_v4: SocketAddrV4,
     addr_v6: SocketAddrV6,
     secret_key: SecretKey,
     rpc_endpoint: IrohServerEndpoint,
     rpc_addr: Option<SocketAddr>,
-    blobs_store: D,
     keylog: bool,
     relay_mode: RelayMode,
-    gc_policy: GcPolicy,
     dns_resolver: Option<DnsResolver>,
     node_discovery: DiscoveryConfig,
-    docs_storage: DocsStorage,
     #[cfg(any(test, feature = "test-utils"))]
     insecure_skip_relay_cert_verify: bool,
```

Comment:
This mirrors all the options of the endpoint API. I found that somewhat tedious. Have you considered starting this builder with an endpoint builder?

```rust
let ep_builder = Endpoint::builder().insecure_skip_relay_cert_verify(true); // do not not call build yet
Node::builder(ep_builder).node_specific_stuff().build().await?;
```